inloop/tasks/admin.py

```diff
@@ -1,10 +1,84 @@
+from shutil import make_archive
+
+from django.conf.urls import url
 from django.contrib import admin
+from django.contrib.auth.models import User
+from django.http import HttpResponse
 
+from inloop.grading.copypasta import jplag_check
+from inloop.grading.models import PlagiarismTest
+from inloop.settings import JPLAG_OUTPUT_PATH
 from inloop.tasks.models import Category, Task
 
 
+def jplag_check_tasks(modeladmin, request, queryset):
+    """
+    Synchronous function to check selected
+    tasks in admin model with JPlag.
+    """
+    users = User.objects.all()
+    jplag_check(users=users, tasks=queryset)
+
+
+def jplag_check_category(modeladmin, request, queryset):
+    """
+    Synchronous convenience function to check selected topics
+    in the admin model with JPlag.
+    """
+    tasks = Task.objects.filter(category__in=queryset)
+    jplag_check_tasks(modeladmin, request, tasks)
+
+
+jplag_check_tasks.short_description = 'Check tasks for Plagiarisms'
+jplag_check_category.short_description = 'Check category for Plagiarisms'
+
+
+def download_jplag_archive(request):
+    """
+    Create a downloadable archive and zip it into
+    a downloadable Response.
+    """
+    path_to_zip = make_archive(JPLAG_OUTPUT_PATH, "zip", JPLAG_OUTPUT_PATH)
```

Comment:
As discussed in our meeting, the download zip function should be implemented as a list change action, and the zip files should be chached by the model.